tests/pipeline/test_factor.py

```diff
@@ -585,6 +586,37 @@ def test_returns(self, seed_value, window_length):
 
         check_allclose(expected, out)
 
+    @parameterized.expand([
+        (100, 15),
+        (101, 4),
+        (102, 100),
+        ])
+    def test_percentchange(self, seed_value, window_length):
+
+        pct_change = PercentChange(
+            inputs=(),
+            window_length=window_length,
+        )
+
+        today = datetime64(1, 'ns')
+        assets = arange(8)
+
+        seed(seed_value)  # Seed so we get deterministic results.
+        test_data = randn(window_length, 8)
+        test_data[0] = array([1, 2, 2, 1, -1, -1, 0, nan])
+        test_data[-1] = array([2, 1, 2, -2, 2, -2, 1, 1])
+
+        # Calculate the expected percent change
+        expected = array([1, -0.5, 0, -3, 3, -1, inf, nan])
+
+        out = empty((8,), dtype=float)
+        pct_change.compute(today, assets, out, test_data)
+
+        check_allclose(expected, out)
+
+        with self.assertRaises(ValueError):
+            PercentChange(inputs=(), window_length=1)
```

Comment:
If we do https://github.com/quantopian/zipline/pull/2506/files#r305114758, we'll probably want to update this check to separately check empty inputs and too-short window_length (as-is, this check will pass, but will only verify whichever of the checks happens to happen first).